### PR TITLE
Scaffold out initial network setup

### PR DESCRIPTION
**Pull request by dsifford**
Including the obvious, this PR does the following things:

1. Adjust the default listening port of the web service AKA "middleware".
2. Add a `.prettierrc.toml` so we can enforce code formatting (prettier to follow in separate PR).
3. Reformatted my code in `processor/`.
4. Scaffolds out the initial working network and infrastructure on Docker + Docker Compose.

Let me know if you have any questions.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
+version: '3.7'
+
+services:
+  validator:
+    image: hyperledger/sawtooth-validator:1.1
+    ports:
+      - '4004:4004'
```
Comment (ej3076) [inline docker-compose.yml:7]: The port standard for sawtooth is interesting - "4004" and "8008"?

File: processor/src/constants.js [hunk 2]
```diff
@@ -17,7 +17,7 @@ module.exports = {
    */
   addressFromVIN(vin) {
     return `${VEHICLE_FAMILY}${hash(vin)}`;
-  }
+  },
```
Comment (ej3076) [inline processor/src/constants.js:20]: Eeek, unnecessary commas!

File: processor/src/state.js [hunk 3]
```diff
@@ -37,16 +37,16 @@ module.exports = class VehicleState {
...
-   * @returns {string[]} The address(es) successfully set.
+   * @returns {Promise<string[]>} The address(es) successfully set.
    */
   async setVehicle(vin, data) {
     const address = addressFromVIN(vin);
     const vehicle = await this._loadVehicle(vin);
     return this.context.setState(
       {
-        [address]: this._serialize(data)
+        [address]: this._serialize(data),
```
Comment (ej3076) [inline processor/src/state.js:47]: Unnecessary comma!!

File: processor/src/state.js [hunk 4]
```diff
@@ -37,16 +37,16 @@ module.exports = class VehicleState {
...
       },
-      this.timeout
+      this.timeout,
```
Comment (ej3076) [inline processor/src/state.js:49]: Another one!!!!

**Review by ej3076 — CHANGES_REQUESTED** [review]
Couple of questions and concerns

**Review by ej3076 — APPROVED** [review]
After discussing - 4004 is the defacto Sawtooth port. Trailing commas are auto-generated for the sake of Git history.

All is well. 🇺🇬

File: processor/src/constants.js (hunk 2 above)
Comment (dsifford) [reply processor/src/constants.js:20]: The most necessary comma I've ever seen

File: processor/src/state.js (hunk 3 above)
Comment (dsifford) [reply processor/src/state.js:47]: The second most necessary comma I've ever seen

File: processor/src/state.js (hunk 4 above)
Comment (dsifford) [reply processor/src/state.js:49]: The third most necessary comma I've ever seen

**Pull request merged**